### DISCUSSION TITLE
Attrs for default coordinate arrays - confusing - intended?

### DISCUSSION THREAD

**Discussion by matthew-brett** [pydata/xarray#8567] [Q&A] [answered]
Sorry for the introductory question, but we (@ivanov and I) ran into this behavior while experimenting:

```python
import numpy as np
data = np.zeros((3, 4, 5))
ds = xr.DataArray(data, dims=('i', 'j', 'k'))
print(ds['k'].attrs)
```

This shows `{}` as we might reasonably expect.  But then:

```python
ds['k'].attrs['foo'] = 'bar'
print(ds['k'].attrs)
```

This also gives `{}`, which we found surprising.   We worked out why that was, after a little experimentation (the default coordinate arrays seems to get created on the fly and garbage collected immediately).  But it took us a little while.   Is that as intended?  Is there a way of making this less confusing?

Thanks for any help.

**Comment by dcherian** [comment, marked as answer]
>  Is that as intended? 

Yes, there is no underlying data stored. 

>  Is there a way of making this less confusing?

I think we should raise an error on that attribute assignment. The message would recommend assigning values to that coordinate first `ds['k'] = np.arange(ds.sizes['k'])`

**Reply by matthew-brett** [reply]
Yes, thanks - an error would have been easier to understand.
